Source/ASCollectionViewProtocols.h

```diff
@@ -10,6 +10,44 @@
 #import <UIKit/UIKit.h>
 #import <AsyncDisplayKit/ASBaseDefines.h>
 
+typedef NS_OPTIONS(NSUInteger, ASCellLayoutMode) {
+  /**
+   * No options set. If cell layout mode is set to ASCellLayoutModeNone, the default values for
+   * each flag listed below is used.
+   */
+  ASCellLayoutModeNone = 0,
+  /**
+   * If ASCellLayoutModeAlwaysSync is enabled it will cause the ASDataController to wait on the
+   * background queue, and this ensures that any new / changed cells are in the hierarchy by the
+   * very next CATransaction / frame draw.
+   *
+   * Note: Sync & Async flags force the behavior to be always one or the other, regardless of the
+   * items. Default: If neither ASCellLayoutModeAlwaysSync or ASCellLayoutModeAlwaysAsync is set,
+   * default behavior is synchronous when there are 0 or 1 ASCellNodes in the data source, and
+   * asynchronous when there are 2 or more.
+  */
+  ASCellLayoutModeAlwaysSync = 1 << 1,                // Default OFF
+  ASCellLayoutModeAlwaysAsync = 1 << 2,               // Default OFF
+  ASCellLayoutModeForceIfNeeded = 1 << 3,             // Deprecated, default OFF.
+  ASCellLayoutModeAlwaysPassthroughDelegate = 1 << 4, // Deprecated, default ON.
+  /** Instead of using performBatchUpdates: prefer using reloadData for changes for collection view */
+  ASCellLayoutModeAlwaysReloadData = 1 << 5,          // Default OFF
+  /** If flag is enabled nodes are *not* gonna be range managed. */
+  ASCellLayoutModeDisableRangeController = 1 << 6,    // Default OFF
+  ASCellLayoutModeAlwaysLazy = 1 << 7,                // Deprecated, default OFF.
```

Comment:
It's strange that we need to add some flags that are deprecated right off the bat. Do we need them at all?

Reply:
Yeah we need them for now at least. We will clean that up eventually.